Sources/Bow/Data/Const.swift

```diff
@@ -35,8 +35,8 @@ public class Const<A, T> : ConstOf<A, T> {
         return Const<A, T>(semigroup.combine(self.value, other.value))
     }
     
-    public func ap<U, SemiG>(_ ff : Const<A, (T) -> U>, _ semigroup : SemiG) -> Const<A, U> where SemiG : Semigroup, SemiG.A == A {
-        return ff.retag().combine(self.retag(), semigroup)
+    public func ap<AA, U, SemiG>(_ fa : Const<A, AA>, _ semigroup : SemiG) -> Const<A, U> where SemiG : Semigroup, SemiG.A == A, T == (AA) -> U {
+        return self.retag().combine(fa.retag(), semigroup)
```

Comment:
In some cases, you add `self` (in the same context - optional) and in the majority, you do not do it. 

Do you think we can unify it, erase them?